cli/flox/doc/manifest.toml.md

```diff
@@ -0,0 +1,310 @@
+    priorities.
+
+## `[vars]`
+
+The `[vars]` section allows you to set environment variables for your
+environment that are set during environment activation.
+The environment variables specified here should not reference one another.
+The names and values of the environment variables are copied verbatim into the
+activation script,
+so capitalization will be preserved.
+
+Example:
+```toml
+[vars]
+DB_URL = ":memory:"
+SERVER_PORT = "3000"
+```
+
+## `[hook]`
+
+The `[hook]` section of the manifest allows you to specify a script that's
+executed immediately after the environment is activated.
+Since the hook runs after activation the environment variables in the `[vars]`
+section may be referenced within the hook.
+
+Common usages for environment hooks are printing usage messages or performing
+setup operations such as initializing a database or starting a server.
+
+The script is set using the `script` option.
+
+```toml
+[hook]
+script = """
+    # Start the development server
+    start_server --port "$SERVER_PORT" --db-url "$DB_URL"
+
+    echo "Server started on port $SERVER_PORT"
+"""
+```
+
+## `[options]`
+
+The `[options]` section of the manifest details settings for the environment
+itself.
+
+The most common option to set is `systems`,
+which specifies which systems that the environment supports.
+A user that attempts to pull an environment from FloxHub when their environment
+isn't explicitly supported will be prompted whether to automatically add their
+system to this list.
+See [`flox-pull(1)`](./flox-pull.md) for more details.
+
+The full set of options are listed below:
+```
+Options ::= {
+  systems                   = null | [<STRING>, ...]
+, allow                     = null | Allows
+, semver                    = null | Semver
+}
+
+Allows ::= {
+  unfree   = null | <BOOL>
+, broken   = null | <BOOL>
+, licenses = null | [<STRING>, ...]
+}
+
+Semver ::= {
+  prefer-pre-releases = <BOOL>
+}
+```
+
+`systems`
+:   The whitelist of systems that this environment supports.
+    Valid values are `x86_64-linux`, `aarch64-linux`,
+    `x86_64-darwin`, and `aarch64-darwin`.
+    [`flox init`](./flox-init.md) automatically populates this list with the
+    current system type.
+
+`allow.unfree`
+:   Allows packages with unfree licenses to be installed and appear in search
+    results.
+    The default is `false`.
+
+`allow.broken`
+:   Allows packages that are marked `broken` in the catalog to be installed and
+    appear in search results.
+    The default is `false`.
+
+`allow.licenses`
+:   A whitelist of software licenses to allow in search results in installs.
+    Valid entries are [SPDX Identifiers](https://spdx.org/licenses).
+
+`semver.prefer-pre-releases`
+:   Whether to prefer pre-release software over stable versions for the
+    purposes of search results and package installations.
+    The default is `false`.
+    Setting this value to `true` would prefer a package version `4.2.0-pre`
+    over `4.1.9`.
```

Comment:
question nonblocking: are we deliberately not mentioning some options, e.g. `package-grouping-strategy`? Is that unimplemented?

Reply:
iirc it was intended as an A/B testing option only

Reply:
Correct, there's a couple of options that I don't think need to be user-facing. `package-repository` is another one.